### ISSUE TITLE
Packaging

### ISSUE THREAD

**Issue by samuelweiler** [w3c/strategy#96]
Jeff Yasskin is proposing new web packaging work at the IETF - see presentation to the DISPATCH WG at IETF99 (July 2017).  https://github.com/WICG/webpackage

**Comment by samuelweiler** [comment]
Based on feedback from IETF99, Jeffrey published two new docs (also visible at the link above):
 
* [Use cases](https://tools.ietf.org/html/draft-yasskin-webpackage-use-cases)
* [Requirements on individual origin-signed request/response pairs](https://tools.ietf.org/html/draft-yasskin-http-origin-signed-responses):
  This is one layer split out of the full packaging proposal. The other layer,
  bundling these signed responses, isn't drafted yet.

**Comment by dontcallmedom** [comment]
this is being tied to the #44 AMP discussion with the recent announcement https://amphtml.wordpress.com/2018/01/09/improving-urls-for-amp-pages/amp/

**Comment by torgo** [comment]
Related TAG Issue: https://github.com/w3ctag/design-reviews/issues/235

**Comment by swickr** [comment]
Related: [Packaging on the Web](https://www.w3.org/TR/2015/WD-web-packaging-20150115/), TAG First Public Working Draft 15 January 2015 [retired]

**Comment by samuelweiler** [comment]
IETF mailing list: https://www.ietf.org/mailman/listinfo/wpack
Slideware for intro that was NOT given at IETF 102 (Montreal, July 2018): https://github.com/danyork/wpack-intro-ietf102

**Comment by sideshowbarker** [comment]
Related:
#### Signed HTTP Exchanges

https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html
https://tools.ietf.org/html/draft-yasskin-http-origin-signed-responses

> Allows sites to send HTTP request+response pairs (exchanges) that are authoritative for an origin, even when the server itself is not authoritative for that origin.
> 
> This is part of Web Packaging, which will allow people to share web applications peer-to-peer, while offline, with a proof the app comes from its original author.

**Comment by sideshowbarker** [comment]
Response today from Jeffrey Yaskin to Mozilla feedback:

https://github.com/mozilla/standards-positions/issues/29#issuecomment-459547918

**Comment by samuelweiler** [comment]
Updated state copied from jyasskin's message to the IETF dispatch list: https://mailarchive.ietf.org/arch/msg/dispatch/e6ZSh7ocW6n2afnbwGzBlX1WT8Y

TL;DR: I'd like to request another discussion of Web Packaging at IETF104. We think it'll be ready for a BOF at IETF105, but mnot suggested we take it back here for a second look.

We dispatched Web Packaging, https://github.com/WICG/webpackage/, at IETF99, with the discussion at https://mailarchive.ietf.org/arch/msg/dispatch/lZohQ8ypodcpB2f_jhxoBRMYl0o. On your advice, we split the proposal into two pieces, "signed exchanges", and "bundles". We also created the https://mailarchive.ietf.org/arch/browse/wpack/ mailing list, although it hasn't seen as much activity as the github issue tracker.

The signed exchanges specification is mostly complete, in an IETF draft that specifies the format: https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html, and a WICG draft that specifies how it integrates with Fetch: https://wicg.github.io/webpackage/loading.html. We've implemented that specification in Chromium, run an origin trial in Chrome, and will be shipping the current version in Chrome 73: https://groups.google.com/a/chromium.org/d/topic/blink-dev/gPH_BcOBEtc/discussion .

Despite shipping the current version, we expect standards bodies to make more changes, and we've built in a way to migrate users to new versions of the specification.

The Bundles specification is very roughly sketched, but we expect to flesh that out and start implementing it in the next few months.

The W3C TAG recently discussed the state of things at https://github.com/w3ctag/meetings/blob/gh-pages/2019/02-tokyo/02-06-minutes.md#signed-exchanges-and-bundling .

Mark Nottingham and a representative of the TAG are organizing an "ESCAPE" conference at a TBD date to help publishers organize their feedback on the proposals. This has been pushed back a couple times, but it sounds like it's more solidly planned for either May or July before IETF105.

We think we'll be able to write a WG charter that we can discuss in a BOF at IETF105, with the goal of handing control of the specifications to a more formal group. However, mnot suggested that we come back to DISPATCH to see if anyone's sense of the right plan has changed now that we have a little more experience with the proposals.

**Comment by samuelweiler** [comment]
Related(?) workshop: https://www.iab.org/2019/05/02/call-for-papers-escape-workshop/

**Comment by samuelweiler** [comment]
IETF WPACK group: https://tools.ietf.org/wg/wpack/
